ontology/disease_syn_enrich.py

```diff
@@ -0,0 +1,188 @@
+#script for enriching disease terms with synonyms
+import os
+import requests
+import json
+import re
+import pandas as pd
+import mysql.connector as mysql
+import joblib
+import numpy as np
+from dotenv import load_dotenv
+
+#loading SQL table
+load_dotenv('.env')
+
+user = os.environ.get("mysql_user")
+pw=os.environ.get("mysql_pwd")
+db=os.environ.get("mysql_db")
+host= "localhost"
+
+conn = mysql.connect(host=host, user=user, passwd=pw, database=db)
+
+#executing the mysql query to filter out the rows
+c = conn.cursor()
+c.execute("select * from disease__nodes;")
+data = c.fetchall()
+column_names = [desc[0] for desc in c.description]
+#converting the table obtained to a dataframe
+df = pd.DataFrame(data, columns= column_names)
+#print (df)
+
+#sending the api request
+def post_request(term, p):
+
+    api_endpoint = "http://data.bioontology.org"
+    endpoint='search'
+    method = 'POST'
+
+    API_KEY = os.environ.get("api_key")
+    
+    kwargs = {
+        'url': api_endpoint+"/"+endpoint+"/?q="+term + "&include=synonym" + "&require_exact_match=true" + "&page="+ str(p),
+        'method': method,
+        'headers': {
+            'Authorization': 'apikey token='+API_KEY
+        }
+    }
+
+    resp = requests.request(**kwargs)
+
+    if int(resp.status_code)!=200:
+        print(resp.text)
+
+    if resp.text:
+        resp_dict = json.loads(resp.text)
+
+    return resp_dict
+
+#extracting only the synonyms from the json file that was obtained as a api request result
+def synonym(resp_dict,i):
+    for x in resp_dict['collection']:
+      str_to_skip = ':'
+      identifier = ''
+      if 'synonym' in x:
+        s = x.get('synonym', [])
+        if not any([str_to_skip.lower() in word.lower() for word in s]):
+          syns[i]=s 
+    return(syns)
+
+
+
+def process_disease_term(disease_term):
+    syns = {}
+    p = 1
+    resp_dict = post_request(disease_term, p)
+    
+    # Looping through each page of the API result
+    for x in resp_dict:
+        if 'pageCount' in x:
+            page = resp_dict.get("pageCount")
+            next = resp_dict.get("nextPage")
+            
+            if page > 1 and next is not None:
+                for p in range(2, page + 1):
+                    resp_dict = post_request(disease_term, p)
+                    syns = synonym(resp_dict, disease_term)
+            else:
+                syns = synonym(resp_dict, disease_term)
+    
+    return syns
+
+def parallel_process_disease(disease_list):
+# Use joblib to parallelize the execution of process_disease
+   syns_list = joblib.Parallel(n_jobs=-1)(joblib.delayed(process_disease_term)(disease) for disease in disease_list)
+# Combine the results from multiple parallel executions
+   combined_syns = {}
+   for result in syns_list:
+     combined_syns.update(result)
+   return combined_syns
+
+#accessing each disease term for api request
```

Comment:
@NayanikaEL All variable declarations and function calls must happen after all functions are defined.